test/e2e/framework/resource_usage_gatherer.go

```diff
@@ -370,6 +370,31 @@ const (
 	MasterAndDNSNodes NodesSet = 2
 )
 
+// isNodeHaveControlPlanePods returns true if specified node has control plane pods
+// (kube-scheduler and/or kube-controller-manager).
+func isNodeHaveControlPlanePods(c clientset.Interface, nodeName string) (bool, error) {
+	regKubeScheduler := regexp.MustCompile("kube-scheduler-.*")
+	regKubeControllerManager := regexp.MustCompile("kube-controller-manager-.*")
+
+	podList, err := c.CoreV1().Pods(metav1.NamespaceSystem).List(context.TODO(), metav1.ListOptions{})
```

Comment:
If we use a `FieldSelector` here, maybe it will be more efficient.

```go
podList, err := c.CoreV1().Pods(metav1.NamespaceSystem).List(context.TODO(), metav1.ListOptions{
    FieldSelector: fields.OneTermEqualSelector("spec.nodeName", nodeName).String(),
})
```

Reply:
Thanks, that is a nice point.
I will update this soon for that.